### PR TITLE
#59 - Create a group change dialog

### PR DESCRIPTION
**Pull request by ImmortalIdiot**

### CODE REVIEW

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt [hunk 1]
```diff
@@ -2,23 +2,63 @@ package com.immortalidiot.rutlead.presentation.viemodels.main
...
+        data class GroupValidationError(val groupError: String?) : State()
     }
 
     var mutableState = MutableStateFlow<State>(State.Init)
         private set
 
+    private val _uiState = MutableStateFlow(ChangeGroupModel(
+        group = String(),
+        isGroupFieldFocused = false
+    ))
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt:27]: refactor

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt [hunk 2]
```diff
@@ -2,23 +2,63 @@ package com.immortalidiot.rutlead.presentation.viemodels.main
...
+    val uiState: StateFlow<ChangeGroupModel> = _uiState.asStateFlow()
+
+    fun clearErrorStack() {
+        mutableState.value = State.ChangeGroupDialog
+    }
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt:32]:
```suggestion
    fun clearErrorStack() {mutableState.value = State.ChangeGroupDialog }
```

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt [hunk 3]
```diff
@@ -2,23 +2,63 @@ package com.immortalidiot.rutlead.presentation.viemodels.main
...
         mutableState.value = State.LogoutDialog
     }
 
     fun onCancelled() {
         mutableState.value = State.Init
     }
+
+    fun changeGroupDialogVisibility() {
+        mutableState.value = State.ChangeGroupDialog
+    }
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt:44]: again

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt [hunk 4]
```diff
@@ -161,6 +191,128 @@ fun ProfileScreen(
         }
     }
 
+    if (profileState is ProfileScreenViewModel.State.ChangeGroupDialog ||
+        profileState is ProfileScreenViewModel.State.GroupValidationError) {
+        Dialog(
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt:196]: вынести в отдельный файл

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt [hunk 5]
```diff
@@ -161,6 +191,128 @@ fun ProfileScreen(
...
+                usePlatformDefaultWidth = false
+            )
+        ) {
+            Box(
+                modifier = modifier.fillMaxHeight(),
+            ) {
+                Column(
+                    modifier = modifier
+                        .fillMaxWidth(0.8f)
+                        .fillMaxHeight(0.25f)
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt:210]: подумай

**Review by LittlePony00 — APPROVED** [review]

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt [hunk 6]
```diff
@@ -2,23 +2,63 @@ package com.immortalidiot.rutlead.presentation.viemodels.main
...
 import androidx.lifecycle.ViewModel
+import com.immortalidiot.rutlead.ui.models.ChangeGroupModel
+import com.immortalidiot.rutlead.validation.validateGroup
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 
 class ProfileScreenViewModel : ViewModel() {
     @Immutable
```
[the remark this answers is not in the document]
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt]: Done

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt (hunk 1 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt]: Done

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt (hunk 2 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt]: Done

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt (hunk 3 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ProfileScreenViewModel.kt]: Done

File: app/src/main/java/com/immortalidiot/rutlead/ui/models/ChangeGroupModel.kt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package com.immortalidiot.rutlead.ui.models
+
+data class ChangeGroupModel(
+    val group: String,
+    val isGroupFieldFocused: Boolean
+)
```
[the remark this answers is not in the document]
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/ui/models/ChangeGroupModel.kt]: Done

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt (hunk 4 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt:310]: Finally, I did it, thx

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt (hunk 5 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt]: :D
Max values has been removed (now these params are using defalut values)

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt [hunk 8]
```diff
@@ -161,6 +198,236 @@ fun ProfileScreen(
...
+        ) {
+            Box(
+                modifier = modifier
+                    .fillMaxHeight()
+                    .verticalScroll(rememberScrollState()),
+            ) {
+                Column(
+                    modifier = modifier
+                        .heightIn(max = 200.dp)
+                        .widthIn(max = 300.dp)
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt:326]: dimensions?

**Review by LittlePony00 — APPROVED** [review]

**Pull request merged**